rewrite-java/src/main/java/org/openrewrite/java/JavaTemplate.java

```diff
@@ -212,6 +213,25 @@ public Builder contextSensitive() {
             return this;
         }
 
+        /**
+         * In context-free templates involving generic types, the type often cannot be inferred automatically.
+         * <p>
+         * Common examples include:
+         * <ul>
+         *   <li>{@code new ArrayList<>()}</li>
+         *   <li>{@code Collections.emptyList()}</li>
+         *   <li>{@code String::valueOf}</li>
+         * </ul>
+         * In such cases, the type must be specified manually.
+         */
+        public Builder expressionType(String expressionType) {
```

Comment:
I also wonder if any of these names would be better: `expectedType()`, `bindType()`, or `typeConstraint()`?